### DISCUSSION TITLE
Per project base extensions

### DISCUSSION THREAD

**Discussion by trampas** [eclipse-theia/theia#14661] [General]
I was wondering if Theia had a per project based extensions (local extensions)?

For the details, I do embedded development and I would like to have my project, stored in git, to have something like an extensions.json.  Then when that project is loaded in Theia it will configure the extensions (and correct versions of extensions, like npm/xpm) and have those extensions be used locally in that project. 

Ideally when a project is loaded in theia for the first time there would be a setup phase it went through.  Basically using xpack(xpm) it could install development tools then install all the extensions and ideally 3rd party tools like Segger Jlink software.  This way you open a project and theia asks you if you want to the setup, after which you can build  and debug the project.

**Comment by msujew** [comment]
Hey @trampas,

Theia already supports extension recommendations, similar to VS Code, see  [the VS Code documentation](https://code.visualstudio.com/docs/editor/extension-marketplace#_workspace-recommended-extensions). However, these are installed globally, and not only for the specific workspace.

If you have a specific feature proposal, please create a [feature proposal issue](https://github.com/eclipse-theia/theia/issues/new?assignees=&labels=&projects=&template=feature_request.md) for this.
